pkg/lib/cockpit-components-terminal.tsx

```diff
@@ -103,13 +104,107 @@ const themes = {
  *
  * Also it is possible to set up theme by property 'theme'.
  */
-export class Terminal extends React.Component {
-    constructor(props) {
+
+export class TerminalState {
+    terminal: Term;
+    element: HTMLDivElement;
+    channel: cockpit.Channel<string> | null;
+
+    #default_channel_creator: (() => cockpit.Channel<string>) | undefined;
+
+    constructor(default_channel_creator?: () => cockpit.Channel<string>) {
+        this.terminal = new Term({
+            cols: 1,
+            rows: 1,
+            cursorBlink: true,
+            fontSize: 16,
+            fontFamily: 'Menlo, Monaco, Consolas, monospace',
+            screenReaderMode: true,
+        });
+        this.terminal.loadAddon(new CanvasAddon());
+        this.element = document.createElement("div");
+        this.channel = null;
+        this.#default_channel_creator = default_channel_creator;
+    }
+
+    connectChannel(channel: cockpit.Channel<string> | undefined) {
+        if (!channel && !this.channel && this.#default_channel_creator)
+            channel = this.#default_channel_creator();
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-21680-5fcf8444-20250814-082605-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)